test/e2e/haproxy_timeouts_test.go

```diff
@@ -0,0 +1,150 @@
+		tlsInspectDelayInput   = 720 * time.Hour
+		tlsInspectDelayOutput  = "2147483647ms" // 720h is greater than the maximum timeout, so it should get clipped to max
+	)
+	icName := types.NamespacedName{Namespace: operatorNamespace, Name: "haproxy-timeout"}
+	domain := icName.Name + "." + dnsConfig.Spec.BaseDomain
+	ic := newPrivateController(icName, domain)
+	ic.Spec.TuningOptions = operatorv1.IngressControllerTuningOptions{
+		ClientTimeout:    &metav1.Duration{Duration: clientTimeoutInput},
+		ClientFinTimeout: &metav1.Duration{Duration: clientFinTimeoutInput},
+		ServerTimeout:    &metav1.Duration{Duration: serverTimeoutInput},
+		ServerFinTimeout: &metav1.Duration{Duration: serverFinTimeoutInput},
+		TunnelTimeout:    &metav1.Duration{Duration: tunnelTimeoutInput},
+		TLSInspectDelay:  &metav1.Duration{Duration: tlsInspectDelayInput},
+	}
+	if err := kclient.Create(context.TODO(), ic); err != nil {
+		t.Fatalf("failed to create ingresscontroller %s: %v", icName, err)
+	}
+	defer assertIngressControllerDeleted(t, kclient, ic)
+	conditions := []operatorv1.OperatorCondition{
+		{Type: operatorv1.IngressControllerAvailableConditionType, Status: operatorv1.ConditionTrue},
+		{Type: operatorv1.LoadBalancerManagedIngressConditionType, Status: operatorv1.ConditionFalse},
+		{Type: operatorv1.DNSManagedIngressConditionType, Status: operatorv1.ConditionFalse},
+	}
+	if err := waitForIngressControllerCondition(t, kclient, 5*time.Minute, icName, conditions...); err != nil {
+		t.Fatalf("failed to observe expected conditions: %v", err)
+	}
+
+	deployment := &appsv1.Deployment{}
+	if err := kclient.Get(context.TODO(), controller.RouterDeploymentName(ic), deployment); err != nil {
+		t.Fatalf("failed to get ingresscontroller deployment: %v", err)
+	}
+	for _, envVar := range deployment.Spec.Template.Spec.Containers[0].Env {
+		switch envVar.Name {
+		case "ROUTER_DEFAULT_CLIENT_TIMEOUT":
+			if envVar.Value != clientTimeoutOutput {
+				t.Errorf("expected %s = %q, got %q", envVar.Name, clientTimeoutOutput, envVar.Value)
+			}
+		case "ROUTER_CLIENT_FIN_TIMEOUT":
+			if envVar.Value != clientFinTimeoutOutput {
+				t.Errorf("expected %s = %q, got %q", envVar.Name, clientFinTimeoutOutput, envVar.Value)
+			}
+		case "ROUTER_DEFAULT_SERVER_TIMEOUT":
+			if envVar.Value != serverTimeoutOutput {
+				t.Errorf("expected %s = %q, got %q", envVar.Name, serverTimeoutOutput, envVar.Value)
+			}
+		case "ROUTER_DEFAULT_SERVER_FIN_TIMEOUT":
+			if envVar.Value != serverFinTimeoutOutput {
+				t.Errorf("expected %s = %q, got %q", envVar.Name, serverFinTimeoutOutput, envVar.Value)
+			}
+		case "ROUTER_DEFAULT_TUNNEL_TIMEOUT":
+			if envVar.Value != tunnelTimeoutOutput {
+				t.Errorf("expected %s = %q, got %q", envVar.Name, tunnelTimeoutOutput, envVar.Value)
+			}
+		case "ROUTER_INSPECT_DELAY":
+			if envVar.Value != tlsInspectDelayOutput {
+				t.Errorf("expected %s = %q, got %q", envVar.Name, tlsInspectDelayOutput, envVar.Value)
+			}
+		}
+	}
```

Comment:
It would be best to actually see these show up in the haproxy.config file with the values that you specified.  You would need to scan a router pod's haproxy.config for the timeout values that should be there.

Reply:
Good point. I will add checks for the values in haproxy's config.